### PR TITLE
feat: cheque commands

### PR DESCRIPTION
**Pull request by Cafe137**

### CODE REVIEW

File: src/command/cheque/deposit.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    super.init()
+
+    if (!(await this.checkDebugApiHealth())) {
+      return
+    }
+
+    const amount = this.parsePositiveBigInt(this.amount)
+
+    if (!amount) {
+      this.console.error('Could not parse amount. Is it a positive number?')
```
Comment (agazso) [inline src/command/cheque/deposit.ts:32]: A better message could be:
`Invalid amount. The amount must be a positive integer.`

File: src/command/cheque/withdraw.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    super.init()
+
+    if (!(await this.checkDebugApiHealth())) {
+      return
+    }
+
+    const amount = this.parsePositiveBigInt(this.amount)
+
+    if (!amount) {
+      this.console.error('Could not parse amount. Is it a positive number?')
```
Comment (agazso) [inline src/command/cheque/withdraw.ts:32]: Same as my other comment

File: test/http-mock/cheque-mock.ts [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+  chequebook: '0x7789f00c8cc0aaa10d2fc50d9c7d74383fd9aa4d',
+  cumulativePayout: 0,
+  beneficiary: '0x1a66fd26f38819a96ca24404ea50a601fa1cb0c3',
+  transactionHash: '0x11df9811dc8caaa1ff4389503f2493a8c46b30c0a0b5f8aa54adbb965374c0ae',
+  result: { recipient: '0x4f16b90d43aa1f9fd3c2fe17dacd9b21417a4134', lastPayout: 0, bounced: false },
+}
+
+const balance = { totalBalance: 100026853000000000, availableBalance: 100018560000000000 }
+
+if (process.argv[2] === 'run') {
```
Comment (nugaon) [inline test/http-mock/cheque-mock.ts:83]: Can you give me a little context about it?

File: src/command/cheque/cheque-command.ts [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+    }
+  }
+
+  private async getCumulativePayout(address: string): Promise<number> {
+    const lastCheques = await this.beeDebug.getLastChequesForPeer(address)
+
+    return lastCheques.lastreceived.payout
+  }
+
+  private async getLastCashedPayout(address: string): Promise<number> {
```
Comment (nugaon) [inline src/command/cheque/cheque-command.ts:87]: can you add a JSDoc for this function please? The Bee API endpoints are not really straightforward here unfortunately.

File: src/command/cheque/cheque-command.ts [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+        return null
+      }
+
+      return bigInt
+    } catch {
+      return null
+    }
+  }
+
+  private async getCumulativePayout(address: string): Promise<number> {
```
Comment (nugaon) [inline src/command/cheque/cheque-command.ts:81]: can you add a JSDoc for this function please? The Bee API endpoints are not really straightforward here unfortunately.

**Review by nugaon — APPROVED** [review]
LGTM, though I can't try it out in "real" environment until I don't finish the test environment with test blockchain.

File: test/http-mock/cheque-mock.ts (hunk 3 above)
Comment (Cafe137) [reply test/http-mock/cheque-mock.ts:83]: There is a JSDoc which I think explains this behaviour.

```
/**
 * Starts a mock HTTP server with predefined cheque responses.
 *
 * Can be run in standalone mode:
 * ts-node test/http-mock/cheque-mock.ts run 1333
 */
```

Is this sufficient?

Comment (nugaon) [reply test/http-mock/cheque-mock.ts:83]: perfect : )

**Pull request merged**